### ISSUE TITLE
Dynamic configuration - remove keys from message

### ISSUE THREAD

**Issue by sazo** [uken/fluent-plugin-elasticsearch#134]
First of all! I love the new elasticsearch_dynamic, it works great and gives so much power to the developer to create dynamic indexes and types without contact to the devOps to change the config of fluentd. :)

Problem : The information about the Index and Type is redundant after usage. FX 

**Fluentd config**
```
type elasticsearch_dynamic
host elasticsearch
port 9200
type_name ${record['type']}
logstash_prefix ${record['index']}
logstash_dateformat %Y.%V
logstash_format true
```

If you are sending this JSON to fluentd.
```json
{
  "index" : "action",
  "type" : "user",
  "message" : "Adding a user"
}
```

ES will automatically add the fields "_index" and "_type" when you search for the document in ES

```json
{
  "index" : "action",
  "_index" : "action-2015.44",
  "type" : "user",
  "_type" : "user",
  "message" : "Adding a user"
}
```

Proposal : 
Make 3 configuration parameters.

```
dynamic_index_record index // will look in the root of the document for the property "index"
dynamic_type_record type // will look in the root of the document for the property "type"
remove_dynamic_vars true // will remove the property defined in "dynamic_index_record" and "dynamic_type_record" before sending it to ES. 
```

**Comment by pitr** [comment]
Keeping in super-configurable spirit, how about something like

```
post_process "record.reject { |k,v| k == 'type' }"
```

**Comment by sazo** [comment]
It will do the job and is more customizable. :) The downside is you need to know about ruby to make it work. 
It all depend on what sprite we are going with. customizable VS easy-to-use. :) Im more easy-to-use guy( FYI i dont know ruby at all :) im php guy)

**Comment by pitr** [comment]
fixed by #164
